lms/djangoapps/teams/serializers.py

```diff
@@ -160,3 +162,6 @@ def __init__(self, *args, **kwargs):
         topics_to_team_count = {d['topic_id']: d['team_count'] for d in teams_per_topic}
         for topic in self.data['results']:
             topic['team_count'] = topics_to_team_count.get(topic['id'], 0)
+
+        if 'sort_order' in self.context and self.context['sort_order'] == 'team_count':
```

Comment:
The order is correct, in that the sort by `team_count` is taking precedence over the alphabetical sort. I think the best solution is the sort by both `name` and `team_count` before paginating. Since this is a little confusing, could you add a test that sorts a list of topics where two topics `aardvarks` and `zebras` have the same team count, but would get put on different pages when sorting alphabetically?